moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h

```diff
@@ -0,0 +1,1099 @@
+            const ros::NodeHandle& node_handle = ros::NodeHandle())
+      : group_name_(group_name), robot_description_(desc), node_handle_(node_handle)
+    {
+    }
+
+    /// The group to construct the class instance for
+    std::string group_name_;
+
+    /// The robot description parameter name (if different from default)
+    std::string robot_description_;
+
+    /// Optionally, an instance of the RobotModel to use can be also specified
+    robot_model::RobotModelConstPtr robot_model_;
+
+    ros::NodeHandle node_handle_;
+  };
+
+  MOVEIT_STRUCT_FORWARD(Plan);
+
+  /// The representation of a motion plan (as ROS messasges)
+  struct Plan
+  {
+    /// The full starting state used for planning
+    moveit_msgs::RobotState start_state_;
+
+    /// The trajectory of the robot (may not contain joints that are the same as for the start_state_)
+    moveit_msgs::RobotTrajectory trajectory_;
+
+    /// The amount of time it took to generate the plan
+    double planning_time_;
+  };
+
+  /**
+      \brief Construct a MoveItCpp instance call using a specified set of options \e opt.
+
+      \param opt. A MoveItCpp::Options structure, if you pass a ros::NodeHandle with a specific callback queue,
+     it has to be of type ros::CallbackQueue
+        (which is the default type of callback queues used in ROS)
+      \param tf_buffer. Specify a TF2_ROS Buffer instance to use. If not specified,
+                        one will be constructed internally along with an internal TF2_ROS TransformListener
+      \param wait_for_servers. Timeout for connecting to action servers. Zero time means unlimited waiting.
+    */
+  MoveItCpp(const Options& opt, const std::shared_ptr<tf2_ros::Buffer>& tf_buffer = std::shared_ptr<tf2_ros::Buffer>(),
+            const ros::WallDuration& wait_for_servers = ros::WallDuration());
+  /**
+      \brief Construct a client for the MoveGroup action for a particular \e group.
+
+      \param tf_buffer. Specify a TF2_ROS Buffer instance to use. If not specified,
+                        one will be constructed internally along with an internal TF2_ROS TransformListener
+      \param wait_for_servers. Timeout for connecting to action servers. Zero time means unlimited waiting.
+    */
+  MoveItCpp(const std::string& group,
+            const std::shared_ptr<tf2_ros::Buffer>& tf_buffer = std::shared_ptr<tf2_ros::Buffer>(),
+            const ros::WallDuration& wait_for_servers = ros::WallDuration());
+
+  ~MoveItCpp();
+
+  /**
+   * @brief This class owns unique resources (e.g. action clients, threads) and its not very
+   * meaningful to copy. Pass by references, move it, or simply create multiple instances where
+   * required.
+   */
+  MoveItCpp(const MoveItCpp&) = delete;
+  MoveItCpp& operator=(const MoveItCpp&) = delete;
+
+  MoveItCpp(MoveItCpp&& other);
+  MoveItCpp& operator=(MoveItCpp&& other);
+
+  /** \brief Get the name of the group this instance operates on */
+  const std::string& getName() const;
```

Comment:
```suggestion
  const std::string& getGroupName() const;
```

Reply:
This is changing the outward API, reducing the immediate ability to just replace MoveGroupInterface with MoveItCpp. I can do this but I don't think it is wise for me to open the "change API" can of worms today.

Reply:
@davetcoleman I'll address this and other changes when fully refactoring the class.